backends/bmv2/jsonconverter.cpp

```diff
@@ -803,13 +805,34 @@ JsonConverter::JsonConverter(const CompilerOptions& options) :
         headerParameter(nullptr), userMetadataParameter(nullptr), stdMetadataParameter(nullptr)
 {}
 
+void addSrcInfoData(Util::JsonObject* result,
```

Comment:
@cc10512: When you say "it will adhere better to the design that we have", do you mean "it will adhere better to the design we hope to have in the near future"?  Because the current code doesn't work as you describe.

It seems I can put in something similar to what I have in this PR now, before that refactor, and let @hanw include these enhancements in his refactoring effort, or I can wait until after that refactoring effort is complete, and do as you suggest.  I suppose a third alternative is to do the refactoring that @hanw is planning to do, but I don't think I want to deny him that pleasure. :-)